lib/ancestry/class_methods.rb

```diff
@@ -202,5 +208,26 @@ def rebuild_depth_cache!
         end
       end
     end
+
+    private
+
+    def sql_concat *parts
+      if ActiveRecord::Base.connection.adapter_name.downcase == 'sqlite'
+        parts.join(' || ')
+      else
+        "CONCAT(#{parts.join(', ')})"
+      end
+    end
+
+    def sql_cast_as_text column
+      #need mysql2 as new rails version support the mysql2 gem only
+      text_type = if ActiveRecord::Base.connection.adapter_name.downcase == 'mysql2'
+        'CHAR'
+      else
+        'TEXT'
+      end
+      # updated mysql thrown error if utf is not defined
+      "CAST(#{column} AS #{text_type}) collate utf8_unicode_ci"
```

Comment:
think the `collate` is better up with mysql only

Reply:
@kbrock why are you even looking at this PR? The original one is #246

Reply:
please be nice